src/Storages/StorageTableFunction.h

```diff
@@ -140,6 +140,13 @@ class StorageTableFunctionProxy final : public StorageProxy
     bool isView() const override { return false; }
     void checkTableCanBeDropped() const override {}
 
+    StoragePolicyPtr getStoragePolicy() const override
+    {
+        if (nested)
+            return StorageProxy::getStoragePolicy();
+        return IStorage::getStoragePolicy();
+    }
```

Comment:
It may work incorrectly, because caller may expect that `getStoragePolicy()` returns actual storage policy of nested storage.

Reply:
(on the other hand, currently there is no way to create a table with non-empty storage policy from table function, so we will never sight that it does not work, but it looks tricky anyway)

Reply:
btw it is already used this way https://github.com/ClickHouse/ClickHouse/blob/307ae344db6335dcdd8363e335650fab8fd482fe/src/Storages/StorageTableFunction.h#L131-L137

Reply:
Not precisely `renameInMemory` has a bit different semantics. If nested storage is already loaded, then update both storage names, but if nested storage is not loaded yet (so we cannot update it), then update proxy storage name and [set](https://github.com/ClickHouse/ClickHouse/blob/2dc5c89b66cff541bb66ee16203b87508e870572/src/Storages/StorageTableFunction.h#L43) nested storage name on future loading.